.github/workflows/release.yml

```diff
@@ -0,0 +1,130 @@
+name: Release
+
+on:
+  push:
+    tags:
+      # For root tags, such as v0.4.2
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-rc.[0-9]+"
+      # For subfolder tags, such as workflow-engine-v1.18.0
+      #- "[a-zA-Z-_]+v[0-9]+.[0-9]+.[0-9]+"
+      #- "[a-zA-Z-_]+v[0-9]+.[0-9]+.[0-9]+-rc.[0-9]+"
+
+jobs:
+  build:
+    permissions:
+      id-token: write
+      contents: read
+      attestations: write
+    strategy:
+      matrix:
+        include:
+          - platform: linux/amd64
+            runner: ubuntu-latest
+            asset_name: ${{ github.event.repository.name }}-linux-amd64-latest
+          - platform: linux/arm64
+            runner: arm-ubuntu-latest-8core
+            asset_name: ${{ github.event.repository.name }}-linux-aarch64-latest
+    runs-on: ${{ matrix.runner }}
+    container: docker-oss.nexus.famedly.de/rust-container:nightly
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Rust
+        uses: famedly/backend-build-workflows/.github/actions/rust-prepare@main
+        with:
+          gitlab_ssh: ${{ secrets.CI_SSH_PRIVATE_KEY}}
+          gitlab_user: ${{ secrets.GITLAB_USER }}
+          gitlab_pass: ${{ secrets.GITLAB_PASS }}
+
+      - name: Caching
+        uses: Swatinem/rust-cache@b8a6852b4f997182bdea832df3f9e153038b5191
+        with:
+          cache-on-failure: true
+          cache-all-crates: true
+
+      - name: Install additional cargo tooling
```

Comment:
We really need to get this stuff into the rust image itself. Not a blocker for this PR, and I guess I'll be doing that with the CI rework anyway.